src/model/backup.ts

```diff
@@ -669,3 +684,132 @@ export async function fetchBackupPassword(): Promise<null | BackupPassword> {
+ *
+ * The goal here is to not allow them into the app if they have broken keychain data from a phone migration
+ *
+ * @returns a promise function to be ran after successful biometric authentication
+ */
+export async function checkIdentifierOnLaunch() {
+  // Unable to really persist things on Android, so let's just exit early...
+  if (IS_ANDROID) return;
+
+  try {
+    const uuid = await getDeviceUUID();
+    if (!uuid) {
+      throw new Error('Unable to retrieve identifier for vendor');
+    }
+
+    const currentIdentifier = await kc.get(identifierForVendorKey);
+    if (currentIdentifier.error) {
+      switch (currentIdentifier.error) {
+        case kc.ErrorType.Unavailable: {
+          logger.debug('Value for current identifier not found, setting it to new UUID...', {
+            uuid,
+            error: currentIdentifier.error,
+          });
+          await kc.set(identifierForVendorKey, uuid);
+          return;
+        }
+
+        case kc.ErrorType.NotAuthenticated: {
+          logger.error(new RainbowError('Not authenticated error while checking identifier on launch'), {
+            error: currentIdentifier.error,
+          });
+          break;
+        }
+
+        case kc.ErrorType.UserCanceled: {
+          logger.error(new RainbowError('User canceled identifier check on launch'), {
+            error: currentIdentifier.error,
+          });
+          break;
+        }
+
+        default:
+          logger.error(new RainbowError('Unknown error while checking identifier on launch'), {
+            error: currentIdentifier.error,
+          });
+          break;
+      }
+
+      throw new Error('Unable to retrieve current identifier');
+    }
+
+    // NOTE: This should never happen, but if we don't have a current identifier, let's set it and exit early
+
+    if (!currentIdentifier.value) {
+      await kc.set(identifierForVendorKey, uuid);
+      return;
+    }
+
+    if (!IS_DEV) {
+      // if our identifiers match up, we can assume no reinstall/migration
+      if (currentIdentifier.value === uuid) {
+        return;
+      }
+    }
+
+    return new Promise(resolve => {
+      Navigation.handleAction(Routes.CHECK_IDENTIFIER_SCREEN, {
+        step: walletBackupStepTypes.check_identifier,
+        // NOTE: Just a reinstall, let's update the identifer and send them back to the app
+        onSuccess: async () => {
+          await kc.set(identifierForVendorKey, uuid);
+          Navigation.goBack();
+          resolve(true);
+        },
+        // NOTE: Detected a phone migration, let's remove keychain keys and send them back to the welcome screen
+        onFailure: async () => {
+          // wipe keychain
+          await kc.clear();
+
+          // re-add the IDFA uuid
+          await kc.set(identifierForVendorKey, uuid);
+
+          // clear async storage
+          await AsyncStorage.clear();
+
+          // clear mmkv
+          clearAllStorages();
+
+          // send user back to welcome screen
```

Comment:
I think it would be nice to let the user know what is going on at least with an alert